// youtube-api-simple/lib/api.js
var Service = require('./service');

module.exports = Api;

/**
 * The api Instance for YouTube
 * @param {Object} options The same object provided for youtube function
 */
function Api(options){
	this.key = options.api;
	this.uri = options.uri;
}

/**
 * The activities service
 * @return {Instance} Instance for activities call to YouTube API
 */
Api.prototype.activities = function(){
	return new Service('activities', this.key);
};

/**
 * The captions service
 * @return {Instance} Instance for captions call to YouTube API
 */
Api.prototype.captions = function(){
	return new Service('captions', this.key);
};

/**
 * The channel sections service
 * @return {Instance} Instance for channel sections call to YouTube API
 */
Api.prototype.channelSections = function(){
	return new Service('channelSections', this.key, this.uri);
};

/**
 * The channels service
 * @return {Instance} Instance for channels call to YouTube API
 */
Api.prototype.channels = function(){
	return new Service('channels', this.key, this.uri);
};

/**
 * The comment threads service
 * @return {Instance} Instance for comment threads call to YouTube API
 */
Api.prototype.commentThreads = function(){
	return new Service('commentThreads', this.key, this.uri);
};

/**
 * The comments service
 * @return {Instance} Instance for comments call to YouTube API
 */
Api.prototype.comments = function(){
	return new Service('comments', this.key, this.uri);
};

/**
 * The fan funding events service
 * @return {Instance} Instance for fan funding events call to YouTube API
 */
Api.prototype.fanFundingEvents = function(){
	return new Service('fanFundingEvents', this.key, this.uri);
};

/**
 * The guide categories service
 * @return {Instance} Instance for guide categories call to YouTube API
 */
Api.prototype.guideCategories = function(){
	return new Service('guideCategories', this.key, this.uri);
};

/**
 * The i18n languages service
 * @return {Instance} Instance for i18n languages call to YouTube API
 */
Api.prototype.i18nLanguages = function(){
	return new Service('i18nLanguages', this.key, this.uri);
};

/**
 * The i18n regions service
 * @return {Instance} Instance for i18n regions call to YouTube API
 */
Api.prototype.i18nRegions = function(){
	return new Service('i18nRegions', this.key, this.uri);
};

/**
 * The live broadcasts service
 * @return {Instance} Instance for live broadcasts call to YouTube API
 */
Api.prototype.liveBroadcasts = function(){
	return new Service('liveBroadcasts', this.key, this.uri);
};

/**
 * The live chat messages service
 * @return {Instance} Instance for live chat messages call to YouTube API
 */
Api.prototype.liveChatMessages = function(){
	return new Service('liveChatMessages', this.key, this.uri);
};

/**
 * The live chat moderators service
 * @return {Instance} Instance for live chat moderators call to YouTube API
 */
Api.prototype.liveChatModerators = function(){
	return new Service('liveChatModerators', this.key, this.uri);
};

/**
 * The live streams service
 * @return {Instance} Instance for live streams call to YouTube API
 */
Api.prototype.liveStreams = function(){
	return new Service('liveStreams', this.key, this.uri);
};

/**
 * The playlists items service
 * @return {Instance} Instance for playlists items call to YouTube API
 */
Api.prototype.playlistItems = function(){
	return new Service('playlistItems', this.key, this.uri);
};

/**
 * The playlists service
 * @return {Instance} Instance for playlists call to YouTube API
 */
Api.prototype.playlists = function(){
	return new Service('playlists', this.key, this.uri);
};

/**
 * The search service
 * @return {Instance} Instance for search call to YouTube API
 */
Api.prototype.search = function(){
	return new Service('search', this.key, this.uri);
};

/**
 * The sponsors service
 * @return {Instance} Instance for sponsors call to YouTube API
 */
Api.prototype.sponsors = function(){
	return new Service('sponsors', this.key, this.uri);
};

/**
 * The subcriptions service
 * @return {Instance} Instance for subcriptions call to YouTube API
 */
Api.prototype.subcriptions = function(){
	return new Service('subcriptions', this.key, this.uri);
};

/**
 * The video abuse report reasons service
 * @return {Instance} Instance for video abuse report reasons call to YouTube API
 */
Api.prototype.videoAbuseReportReasons = function(){
	return new Service('videoAbuseReportReasons', this.key, this.uri);
};

/**
 * The videos categories service
 * @return {Instance} Instance for videos categories call to YouTube API
 */
Api.prototype.videosCategories = function(){
	return new Service('videosCategories', this.key, this.uri);
};

/**
 * The videos service
 * @return {Instance} Instance for videos call to YouTube API
 */
Api.prototype.videos = function(){
	return new Service('videos', this.key, this.uri);
};
